server/src/test/java/io/crate/integrationtests/TransportSQLActionTest.java

```diff
@@ -1999,6 +2003,29 @@ public void test_bit_string_can_be_inserted_and_queried() throws Exception {
         ));
     }
 
+    /**
+     * Adds innerTypes to object type definitions
+     **/
+    private static DataType<?> extendedType(DataType<?> type, Object value) {
+        if (type.id() == ObjectType.ID) {
+            var entryIt = ((Map<?, ?>) value).entrySet().iterator();
+            Builder builder = ObjectType.builder();
+            while (entryIt.hasNext()) {
```

Comment:
Out of curiosity: any reason to use iter instead of for loop?

Reply:
It's a leftover from the first version which generated a string and used a .hasNext inside the loop to add a comma if it's not the last.